### PR TITLE
doc: clarify how Lean supports constructive logic

### PR DESCRIPTION
**Pull request by chabulhwi**
The paragraph I added to Section 'Historical and Philosophical Context'
explains how Lean core [supports constructive logic][0], whether the
Lean core team will provide tactics for constructive logic or [accept
them from contributors][1], and whether a user can [develop them by
oneself][2] [outside the core][3].

I also added a cautionary note for constructivists to Chapter
'Introduction.'

[0]:
https://leanprover.zulipchat.com/#narrow/stream/348111-std4/topic/Movement.20from.20Std.20to.20Init/near/430339840
[1]:
https://leanprover.zulipchat.com/#narrow/stream/348111-std4/topic/How.20classical.20is.20std4.3F/near/383780177
[2]:
https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/constructive.20tactic.20mode.20in.20lean/near/431685357
[3]:
https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/constructive.20tactic.20mode.20in.20lean/near/431714863

Co-authored-by: Mario Carneiro <di.gama@gmail.com>
Co-authored-by: Henrik Böving <hargonix@gmail.com>
Co-authored-by: Kim Morrison <kim@tqft.net>

### CODE REVIEW

File: introduction.md [hunk 1]
```diff
@@ -61,6 +61,15 @@ To install Lean in your computer consider using the [Quickstart](https://github.
 
 This tutorial describes the current version of Lean, known as Lean 4.
 
+Be aware that some tactics in Lean's core library use classical reasoning by
```
Comment (eric-wieser) [inline introduction.md:64]: This definitely doesn't belong here; this is not a section about constructivity.

Comment (chabulhwi) [reply introduction.md:64]: If you want to remove the paragraph from here, I'll do it. But wouldn't constructivists appreciate it if this textbook informed them about the current behaviors of some tactics in Lean core before they begin to read the book?

**Comment by chabulhwi** [comment]
@eric-wieser I added a cautionary note for constructivists to Chapter Introduction.

**Comment by chabulhwi** [comment]
I'll close this pull request since https://github.com/leanprover/theorem_proving_in_lean4/pull/117 will revise the text.

**Pull request closed without merging**